sa/sa_test.go

```diff
@@ -4307,3 +4308,583 @@ func TestGetSerialsByAccount(t *testing.T) {
 	test.AssertNotError(t, err, "calling GetSerialsByAccount")
 	test.AssertEquals(t, len(seen), 2)
 }
+
+func TestUnpauseAccount(t *testing.T) {
+	if os.Getenv("BOULDER_CONFIG_DIR") != "test/config-next" {
+		t.Skip("Test requires paused database table")
+	}
+	sa, _, cleanUp := initSA(t)
+	defer cleanUp()
+
+	ptrTime := func(t time.Time) *time.Time {
+		return &t
+	}
+
+	type args struct {
+		state []pausedModel
+		req   *sapb.RegistrationID
+	}
+	tests := []struct {
+		name string
+		args args
+	}{
+		{
+			name: "UnpauseAccount with no paused identifiers",
+			args: args{
+				state: nil,
+				req:   &sapb.RegistrationID{Id: 1},
+			},
+		},
+		{
+			name: "UnpauseAccount with one paused identifier",
+			args: args{
+				state: []pausedModel{
+					{
+						RegistrationID: 1,
+						identifierModel: identifierModel{
+							Type:  identifierTypeToUint[string(identifier.DNS)],
+							Value: "example.com",
+						},
+						PausedAt: ptrTime(sa.clk.Now().Add(-time.Hour)),
+					},
+				},
+				req: &sapb.RegistrationID{Id: 1},
+			},
+		},
+		{
+			name: "UnpauseAccount with multiple paused identifiers",
+			args: args{
+				state: []pausedModel{
+					{
+						RegistrationID: 1,
+						identifierModel: identifierModel{
+							Type:  identifierTypeToUint[string(identifier.DNS)],
+							Value: "example.com",
+						},
+						PausedAt: ptrTime(sa.clk.Now().Add(-time.Hour)),
+					},
+					{
+						RegistrationID: 1,
+						identifierModel: identifierModel{
+							Type:  identifierTypeToUint[string(identifier.DNS)],
+							Value: "example.net",
+						},
+						PausedAt: ptrTime(sa.clk.Now().Add(-time.Hour)),
+					},
+					{
+						RegistrationID: 1,
+						identifierModel: identifierModel{
+							Type:  identifierTypeToUint[string(identifier.DNS)],
+							Value: "example.org",
+						},
+						PausedAt: ptrTime(sa.clk.Now().Add(-time.Hour)),
+					},
+				},
+				req: &sapb.RegistrationID{Id: 1},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Setup table state.
+			for _, state := range tt.args.state {
+				err := sa.dbMap.Insert(ctx, &state)
+				test.AssertNotError(t, err, "inserting test identifier")
+			}
+
+			_, err := sa.UnpauseAccount(ctx, tt.args.req)
+			test.AssertNotError(t, err, "Unexpected error for UnpauseAccount()")
+
+			// Count the number of paused identifiers.
+			var count int
+			err = sa.dbReadOnlyMap.SelectOne(
+				ctx,
+				&count,
+				"SELECT COUNT(*) FROM paused WHERE registrationID = ? AND unpausedAt IS NULL",
+				tt.args.req.Id,
+			)
+			test.AssertNotError(t, err, "SELECT COUNT(*) failed")
+			test.AssertEquals(t, count, 0)
+
+			// Drop all rows from the paused table.
+			_, err = sa.dbMap.ExecContext(ctx, "TRUNCATE TABLE paused")
```

Comment:
I believe our "test.AssertFoo" functions result in the test _immediately_ terminating, meaning that this cleanup line won't get run. Maybe put it in a defer instead, like how the top level of this test does
```go
sa, fc, cleanup := initSA(t)
defer cleanup()
```

Same in other tests below.